src/markov.py

```diff
@@ -24,11 +25,33 @@ def __init__(self,
         self.trans = trans
 
 
+
 def likelihood(x: list[int], mm: MarkovModel) -> float:
     """
     Compute the likelihood of mm given x.
 
     This is the same as the probability of x given mm,
     i.e., P(x ; mm).
     """
-    ...  # FIXME: implement this
+    for i in range(len(x)): 
+        if i == 0: 
+            prob = mm.init_probs[x[i]]
+        else: 
+            prob = prob * mm.trans[x[i-1]][x[i]]
```

Comment:
If you move the special case out of the loop, you don't have to check the condition every time it runs, and that will speed up the code.

```python
prob = mm.init_probs[x[0]]
for i in range(1, len(x)):
    ...
```